iina/QuickSettingViewController.swift

```diff
@@ -825,26 +845,110 @@ class QuickSettingViewController: NSViewController, NSTableViewDataSource, NSTab
     redraw(indicator: audioDelaySliderIndicator, constraint: audioDelaySliderConstraint, slider: audioDelaySlider, value: "\(sender.stringValue)s")
   }
 
-  @IBAction func audioEqSliderAction(_ sender: NSSlider) {
-    player.setAudioEq(fromGains: [
-      audioEqSlider1.doubleValue,
-      audioEqSlider2.doubleValue,
-      audioEqSlider3.doubleValue,
-      audioEqSlider4.doubleValue,
-      audioEqSlider5.doubleValue,
-      audioEqSlider6.doubleValue,
-      audioEqSlider7.doubleValue,
-      audioEqSlider8.doubleValue,
-      audioEqSlider9.doubleValue,
-      audioEqSlider10.doubleValue,
-      ])
-  }
-
-  @IBAction func resetAudioEqAction(_ sender: AnyObject) {
-    player.removeAudioEqFilter()
-    updateAudioEqState()
+  func applyEQ(_ profile: EQProfile) {
+    zip(eqSliders, profile.gains).forEach { (slider, gain) in
+      slider.doubleValue = gain
+    }
+    player.setAudioEq(fromGains: profile.gains)
+  }
+
+  private func makeProfileNameValidationAlert(isNewProfile: Bool) -> NSAlert {
+    let panel = NSAlert()
+    let textInput = NSTextField(frame: NSRect(x: 0, y: 0, width: 200, height: 24))
+    panel.accessoryView = textInput
+    let firstButton = panel.addButton(withTitle: NSLocalizedString("general.ok", comment: "OK"))
+    firstButton.isEnabled = false
+    panel.addButton(withTitle: NSLocalizedString("general.cancel", comment: "Cancel"))
+    let key = isNewProfile ? "eq.new_profile" : "eq.rename"
+    panel.messageText = NSLocalizedString(key, comment: key)
+    let nameList = eqPopUpButton.itemArray.filter{ $0.tag == 1 || $0.tag == 0 }.map{ $0.title }
+    NotificationCenter.default.addObserver(forName: NSControl.textDidChangeNotification, object: textInput, queue: .main) { _ in
+      let input = textInput.stringValue
+      if input.isEmpty {
+        firstButton.isEnabled = false
+        return
+      }
+      if nameList.contains( where: { $0 == input } ) {
+        firstButton.isEnabled = false
+      } else {
+        firstButton.isEnabled = true
+        self.inputString = input
+      }
+    }
+    return panel
+  }
```

Comment:
We should reuse `Utility.quickPromptPanel()`. You can make it accept a validation block or something like this.

It would be even better if you can add some help text to tell the user why the input is not accepted, with some default messages like "Name already exists", which can also be reused elsewhere.

Or you can leave this to me after merging this PR.

Reply:
There's another bug: you also need to make sure the user defined name is not "Manual" (or localized equivalent).

I'm thinking about whether it's safe to trust localized names in the code. If there are duplicated preset names (whether due to translation mistake or whatever reason), code like `eqPopUpButton.selectItem(withTitle:)` will not work properly. So perhaps it's better to use some internal enum or identifiers.